### DISCUSSION TITLE
Should SQLite be faster than MySQL?

### DISCUSSION THREAD

**Discussion by TannedCung** [milvus-io/milvus#9944] [Q&A and General discussion] [answered]
Hi,

I have just had a small benchmark about milvus and surprisingly, i found that using SQLite improve time cost per query remarkably, even though, MySQl is the recommended storage.

Here and my setting:
- CPU: Intel Xeon E312xx 2.4Ghzx8
- RAM: 12GB
- HDD: 100GB
- UBUNTU: 20.04
- Milvus: 1.0.0_cpu

And Milvus setting:
- Index: IVF_FLAT
- 2 collections (100000 embs of size [1,512] for each collections)
- nprobe: 16
- topk: 1
- nlist: 2048

For the 2 collections i created 1 process for each and search simultaneously with the batch size of 2. I did this twice, 1 for SQLite and the other for MySQL. (MySQL from docker 5.7 as instruction from docs)

And here is the result:

| Process | SQLite | MySQL |
| -- | -- | -- |
| 1 | 0.005390320179518312 | 0.008018158863205463 |
| 2 | 0.005390804626885802 | 0.008025996528659016 |

i wonder if this is a normal result???
Thank for your help !!

**Comment by yhmo** [comment, marked as answer]
SQLite is in-process,  MySQL is TCP/IP connection, so SQLite is faster than MySQL.
In small dataset, the connection between milvus and meta service may become bottleneck. But in large dataset, the time cost in meta service connection is ignore-able.
MySQL is an independent service, is more reliable than in-process service. And most of user case is large dataset(millions of embeddings). This is the reason why we recommend MySQL as meta service.

**Comment by genausz** [comment]
In my case, there are only 10~100k embeddings. Is it okay to use SQLite in production?

**Reply by yhmo** [reply]
> In my case, there are only 10~100k embeddings. Is it okay to use SQLite in production?

10~100k is a really small dataset, SQLite is ok.
